### PR TITLE
LOAD-3218: Add name field for file type

### PR DESCRIPTION
**Pull request by paulsgrudups**

### CODE REVIEW

File: src/main/java/com/loadero/model/File.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import com.google.gson.annotations.SerializedName;
+import com.loadero.Loadero;
+import com.loadero.http.ApiResource;
+import com.loadero.http.RequestMethod;
+import java.io.IOException;
+
+/**
+ * Used to read information about a project file.
+ */
+public final class File {
```
Comment (ajpetersons) [inline src/main/java/com/loadero/model/File.java:12]: i'm thinking that with media file removal we actually already have a breaking change here, so might as well solve this one properly by replacing the script type. is there anything else we might be deprecating in the near future?

File: src/main/java/com/loadero/model/File.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    /**
+     * Retrieves file from Loadero.
+     *
+     * @param fileId ID of the file.
+     * @return {@link File} object.
+     * @throws IOException if request failed.
+     */
+    public static File read(int fileId) throws IOException {
```
Comment (ajpetersons) [inline src/main/java/com/loadero/model/File.java:46]: shouldn't we also add a create method?

**Review by ajpetersons — CHANGES_REQUESTED** [review]

File: src/main/java/com/loadero/model/File.java (hunk 2 above)
Comment (paulsgrudups) [reply src/main/java/com/loadero/model/File.java:42]: I added it, but it fails to get the id. 
`An exception occurred while executing the Java class. Can not get final int field com.loadero.model.File.id on com.loadero.model.FileParams` which for some reason I fail to debug..

File: src/main/java/com/loadero/http/FileDeserializer.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    ) throws JsonParseException {
+        JsonObject jsonObject = json.getAsJsonObject();
+        if (jsonObject == null) {
+            return null;
+        }
+
+        FileType fileType = FileType
+            .getConstant(jsonObject.getAsJsonPrimitive("file_type").getAsString());
+
+            FileParams params = FileParams
```
Comment (ajpetersons) [inline src/main/java/com/loadero/http/FileDeserializer.java:25]:
```suggestion
        FileParams params = FileParams
```

File: src/main/java/com/loadero/model/File.java [hunk 4]
```diff
@@ -84,18 +96,17 @@ public String getPassword() {
...
-            '}';
+                "id=" + id +
+                ", created='" + created + '\'' +
+                ", updated='" + updated + '\'' +
+                ", projectId=" + projectId +
+                ", fileType='" + fileType + '\'' +
+                ", content='" + content + '\'' +
+                ", name='" + name + '\'' +
+                ", password='" + password + '\'' +
+                '}';
```
Comment (ajpetersons) [inline src/main/java/com/loadero/model/File.java:110]: i don't think this should be that indented

File: src/main/java/com/loadero/model/Script.java
Comment (ajpetersons) [inline src/main/java/com/loadero/model/Script.java:1]: update readme:88

File: src/main/java/com/loadero/model/FileParams.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+        this.fileType = builder.fileType;
+        this.content = builder.content;
+        this.name = builder.name;
+        this.password = builder.password;
+    }
+
+    private FileParams(
+            int id, String created, String updated,
+            int projectId, FileType fileType,
+            String content, String name, String password) {
```
Comment (ajpetersons) [inline src/main/java/com/loadero/model/FileParams.java:35]:
```suggestion
            String content, String name, String password
        ) {
```

File: src/main/java/com/loadero/model/FileParams.java [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    FileParams copyUncommonFields(File currentParams) {
+        FileType fileType = this.fileType == null ? currentParams.getFileType() : this.fileType;
+        String name = this.name == null ? currentParams.getName() : this.name;
+        String content = this.content == "" ? currentParams.getContent() : this.content;
+        String password = this.password == "" ? currentParams.getPassword() : this.password;
+
+        return new FileParams(
+                id, created, updated,
+                projectId, fileType, content,
+                name, password);
```
Comment (ajpetersons) [inline src/main/java/com/loadero/model/FileParams.java:65]:
```suggestion
            id, created, updated,
            projectId, fileType, content,
            name, password
        );
```

File: src/main/java/com/loadero/model/FileParams.java [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+        return "FileParams{" +
+                "id=" + id +
+                ", created='" + created + '\'' +
+                ", updated='" + updated + '\'' +
+                ", projectId=" + projectId +
+                ", fileType='" + fileType + '\'' +
+                ", content='" + content + '\'' +
+                ", name='" + name + '\'' +
+                ", password='" + password + '\'' +
+                '}';
```
Comment (ajpetersons) [inline src/main/java/com/loadero/model/FileParams.java:220]:
```suggestion
            "id=" + id +
            ", created='" + created + '\'' +
            ", updated='" + updated + '\'' +
            ", projectId=" + projectId +
            ", fileType='" + fileType + '\'' +
            ", content='" + content + '\'' +
            ", name='" + name + '\'' +
            ", password='" + password + '\'' +
            '}';
```

**Review by ajpetersons — CHANGES_REQUESTED** [review]

**Review by ajpetersons — APPROVED** [review]

**Pull request merged**